protostar/compiler/contract_source_identifier.py

```diff
@@ -0,0 +1,65 @@
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Protocol
+
+from protostar.protostar_exception import ProtostarException
+from protostar.self import ContractName
+
+
+@dataclass(eq=True, frozen=True, unsafe_hash=True)
+class ContractSourceIdentifier:
```

Comment:
This name is confusing

Reply:
This is `ContractSources` or `ContractEntrypoints` but i don't think it's an identifier.

Reply:
I'd avoid plural names. An `entrypoint` suggests `@view` and `@external` functions. I'll change it to `ContractSource`.

```py
contract_sources = ContractSources()
contract_sources # <= the name suggest it's an array
```

Reply:
I changed my mind. `ContractSource` can be interpreted as a contract source code. `ContractSourceIdentifier` identifies source code(s).
I'll leave `ContractSourceIdentifier` unless you have a better name.

Reply:
Yep it definetely is not an `Identifier`, agreed with @Arcticae.

`ContractAlias` IMO